internal/merror/merror.go

```diff
@@ -0,0 +1,37 @@
+package merror
+
+import (
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// MError is metro error
+type MError struct {
```

Comment:
should we add an optional metadata map as well? Sometimes adding too much context in the error message itself will make it verbose. All such extra (k,v) can be parked inside the metadata map to enrich the error.
```
{
	"code": "TopicCreationFailed",
	"message": "failed to created topic",
	"metadata": {
		"topic": "payments.auth",
		"broker": "kafka",
		"some-use-full-key-in-future": "value"
	}
}
```
Thoughts?